tests/lib/pkgdb.sh

```diff
@@ -560,14 +560,14 @@ pkg_dependencies_ubuntu_classic(){
 }
 
 pkg_linux_image_extra(){
-    if apt-cache policy "linux-image-extra-$(uname -r)" >/dev/null 2>&1; then
+    available_pkgs=$(apt-cache policy linux-image-extra-.*-generic | grep -E 'linux-image-extra-.*:' | tr -d :)
+    if echo "$available_pkgs" | grep -q "linux-image-extra-$(uname -r)"; then
         echo "linux-image-extra-$(uname -r)"
-    elif apt-cache policy "linux-image-extra-$(uname -r | cut -d- -f1)" >/dev/null 2>&1; then
-        echo "linux-image-extra-$(uname -r | cut -d- -f1)"
-    else
-        echo "cannot find a matching kernel modules package"
-        exit 1
+        return
     fi
+
+    kernel_version="$(uname -r | cut -d- -f1)"
```

Comment:
Something fishy here. Assuming `uname -r` returns `4.15.0-1017-gcp`, this will get `4.15.0`. So the next line picks `linux-image-extra-4.15.0.*`. With the earlier processing you'd try to match `linux-image-extra-4.15.0-1017-gcp` which will fail, because there is no such package listed here https://packages.ubuntu.com/xenial/kernel/ Then it proceeds to pick the first `linux-image-extra-4.15.0` package from the list, which is not the GCE specific one (because we've checked it does not exist) and you end up switching the kernel. 

FWIW, I see there's a `linux-modules-extra-4.15.0-1017-gcp`, not `linux-image-extra..` for this kernel.

Reply:
Can you try this instead?
```
pkg_linux_image_extra () 
{ 
    if apt-cache show "linux-image-extra-$(uname -r)" > /dev/null 2>&1; then
        echo "linux-image-extra-$(uname -r)";
    else
        if apt-cache show "linux-modules-extra-$(uname -r)" > /dev/null 2>&1; then
            echo "linux-modules-extra-$(uname -r)";
        else
            echo "cannot find a matching kernel modules package";
            exit 1;
        fi;
    fi
}
```

Reply:
@bboozzoo done, thanks !!!